drivers/espi/host_subs_npcx.c

```diff
@@ -128,6 +128,14 @@
 #include <zephyr/irq.h>
 LOG_MODULE_REGISTER(host_sub_npcx, LOG_LEVEL_ERR);
 
+/** Minimal ring buffer implementation */
+struct ring_buf {
+	uint32_t *buf;
+	uint16_t len;
+	uint16_t head;
+	uint16_t tail;
+};
```

Comment:
Could you use the Zephyr ring buffer implementation instead? https://docs.zephyrproject.org/latest/kernel/data_structures/ring_buffers.html

Reply:
+1 for using the Zephyr ring buffer.

Reply:
Done.